### DISCUSSION TITLE
Ref

### DISCUSSION THREAD

**Discussion by imlixixi** [nasa/fprime#1313] [Q&A]
As the Ref/docs/ssd.md  mentioned, I understand the relevant contents in Ref but I still have two questions:
1.How are commands handled inside each component? (eg a component received a command (such as  calculating 1+1) ),  I'm not sure where these processes are implemented.
2.In fprime/Ref/SignalGen directory, I'm not sure whether the file "SignalGen.cpp"   is automatically generated or handwritten？

**Comment by ThibFrgsGmz** [comment]
For part 1.
When defining a component with a FPP model (in v3.0.x) or in XML (in v2.x) you have the possibility to specify its port types defining its afferent and efferent couplings: https://github.com/nasa/fprime/blob/devel/docs/Tutorials/MathComponent/Tutorial.md#41-construct-the-fpp-model

When you follow the construction chain you will get to the step of generating the template of the corresponding C++ source code. 
There you will see the handlers of the ports that you have defined previously, especially the commands: https://github.com/nasa/fprime/blob/devel/docs/Tutorials/MathComponent/Tutorial.md#41-construct-the-fpp-model

Once you have this template, you implement your business logic in the handle to define the behavior of your component once this command is received (1+1 in your example).

**Reply by imlixixi** [reply]
Thank u very much for ur patient answer！

**Reply by ThibFrgsGmz** [reply]
For your second problem, the component SignalGen has a FPP model ( [SignalGen.fpp](https://github.com/nasa/fprime/blob/devel/Ref/SignalGen/SignalGen.fpp), [Commands.fppi](https://github.com/nasa/fprime/blob/devel/Ref/SignalGen/Commands.fppi), [Telemetry.fppi](https://github.com/nasa/fprime/blob/devel/Ref/SignalGen/Telemetry.fppi), [Events.fppi](https://github.com/nasa/fprime/blob/devel/Ref/SignalGen/Events.fppi) ) and this allowed to generate a CPP template containing the handler of the defined ports.

Then they implemented the business logic of the component.
